### PR TITLE
feat(compiler): integrate compiler with view engine - di, query, cont…

### PR DESCRIPTION
**Pull request by tbosch**
…ent projection tests

Included refactoring:
- use a number as query id
- use a bloom filter for aggregating matched queries of nested elements
- separate static vs dynamic queries

Part of #14013

### CODE REVIEW

File: modules/@angular/core/src/view/element.ts [hunk 1]
```diff
@@ -62,8 +75,17 @@ export function elementDef(
   // skip the call to sliceErrorStack itself + the call to this function.
   const source = isDevMode() ? sliceErrorStack(2, 3) : '';
   const matchedQueryDefs: {[queryId: string]: QueryValueType} = {};
+  let matchedQueryIds = 0;
```
Comment (tbosch) [inline modules/@angular/core/src/view/element.ts]: move out into helper and use everywhere.

File: modules/@angular/core/src/view/query.ts [hunk 2]
```diff
@@ -53,103 +56,122 @@ export function createQuery(): QueryList<any> {
...
     }
-    elIndex = viewParentElIndex(view);
-    view = view.parent;
   }
-  if (!view) {
-    throw new Error(
-        `Illegal State: Tried to dirty parent query ${queryId} but the query could not be found!`);
+
+  // view queries
+  let compDef = view.parentNodeDef;
```
Comment (tbosch) [inline modules/@angular/core/src/view/query.ts]: hostDef

File: modules/@angular/core/src/view/query.ts [hunk 3]
```diff
@@ -53,103 +56,122 @@ export function createQuery(): QueryList<any> {
...
       const embeddedViews = elementData.embeddedViews;
-      for (let k = 0; k < embeddedViews.length; k++) {
-        const embeddedView = embeddedViews[k];
-        const dvc = declaredViewContainer(embeddedView);
-        if (dvc && dvc === elementData) {
-          calcQueryValues(embeddedView, 0, embeddedView.def.nodes.length - 1, queryId, values);
+      if (embeddedViews) {
+        for (let k = 0; k < embeddedViews.length; k++) {
+          const embeddedView = embeddedViews[k];
+          // TODO: do we still need to check this??
```
Comment (tbosch) [inline modules/@angular/core/src/view/query.ts]: remove TODO

File: modules/@angular/core/test/linker/entry_components_integration_spec.ts [hunk 4]
```diff
@@ -14,6 +15,19 @@ import {Console} from '../../src/console';
 
 
 export function main() {
+  // describe('Current compiler', () => { createTests({viewEngine: false}); });
```
Comment (tbosch) [inline modules/@angular/core/test/linker/entry_components_integration_spec.ts]: uncomment

File: modules/@angular/core/test/linker/entry_components_integration_spec.ts [hunk 5]
```diff
@@ -14,6 +15,19 @@ import {Console} from '../../src/console';
...
+
+  describe('View Engine compiler', () => {
```
Comment (tbosch) [inline modules/@angular/core/test/linker/entry_components_integration_spec.ts]: also remove viewEngine `xit` in `integration_spec`

File: modules/@angular/core/test/linker/query_integration_spec.ts [hunk 6]
```diff
@@ -267,9 +281,10 @@ export function main() {
       it('should contain the first descendant content child templateRef', () => {
         const template = '<needs-content-child-template-ref-app>' +
             '</needs-content-child-template-ref-app>';
-        const view = createTestCmpAndDetectChanges(MyComp0, template);
+        const view = createTestCmp(MyComp0, template);
```
Comment (tbosch) [inline modules/@angular/core/test/linker/query_integration_spec.ts]: Try to revert, as queryIds are now exact if less than 32.

**Review by vicb — APPROVED** [review]

File: modules/@angular/core/src/view/element.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/src/view/element.ts]: Done

File: modules/@angular/core/src/view/query.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/core/src/view/query.ts]: TODO

File: modules/@angular/core/test/linker/entry_components_integration_spec.ts (hunk 4 above)
Comment (tbosch) [inline modules/@angular/core/test/linker/entry_components_integration_spec.ts]: Done

File: modules/@angular/core/test/linker/entry_components_integration_spec.ts (hunk 5 above)
Comment (tbosch) [inline modules/@angular/core/test/linker/entry_components_integration_spec.ts]: Done

**Review by IgorMinar — APPROVED** [review]

**Pull request merged**